src/environ/_environ_config.py

```diff
@@ -174,11 +194,73 @@ class Sub:
 
     You can nest your configuration as deeply as you wish.
     """
+    default = None if optional else RAISE
     return attr.ib(
-        default=None, metadata={CNF_KEY: _ConfigEntry(None, None, cls, True)}
+        default=default,
+        metadata={CNF_KEY: _ConfigEntry(None, default, cls, True)},
     )
 
 
+def __default_getter(environ, metadata, prefix, name):
```

Comment:
Why the dunder on a function? Also please informal docstring. :)

Reply:
It's *really really* private :laughing: I'll just single underscore this and `__to_config_recurse()`

Reply:
As below, done but I'll let you resolve